src/manifest.c

```diff
@@ -494,14 +514,25 @@ static int try_delta_manifest_download(int current, int new, char *component, st
 }
 
 /* TODO: This should deal with nested manifests better */
-static int retrieve_manifests(int current, int version, char *component, struct file *file)
+static int retrieve_manifests(int current, int version, char *component, char *statepath, struct file *file)
 {
 	char *url = NULL;
 	char *filename;
 	char *dir;
+	char *basedir;
 	int ret = 0;
 	char *tar;
 	struct stat sb;
+	bool is_mix = false;
+
+	if (!statepath) {
+		basedir = state_dir;
+	} else {
+		basedir = statepath;
+	}
```

Comment:
Similar to my question from the code in `manifest_from_file`, is statepath always the same value for the mix case? If so, then I would prefer the API to be consistent and add is_mix as a new parameter to the function instead.